autoPyTorch/api/base_task.py

```diff
@@ -1,139 +1,740 @@
+                The function must accept the arguments scenario_dict,
+                instances, num_params, runhistory, seed and ta. This is
+                an advanced feature. Use only if you are familiar with
+                [SMAC](https://automl.github.io/SMAC3/master/index.html).
+            all_supported_metrics (bool), (default=True): if True, all
+                metrics supporting current task will be calculated
+                for each pipeline and results will be available via cv_results
+            precision (int), (default=32): Numeric precision used when loading
+                ensemble data. Can be either '16', '32' or '64'.
+            disable_file_output (Union[bool, List]):
+            load_models (bool), (default=True): Whether to load the
+                models after fitting AutoPyTorch.
+
+        Returns:
+            self
+
+        """
+        if self.task_type != dataset.task_type:
+            raise ValueError("Incompatible dataset entered for current task,"
+                             "expected dataset to have task type :{} got "
+                             ":{}".format(self.task_type, dataset.task_type))
+
+        # Initialise information needed for the experiment
+        experiment_task_name = 'runSearch'
+        self._dataset_requirements = get_dataset_requirements(
+            info=self._get_required_dataset_properties(dataset))
+        dataset_properties = dataset.get_dataset_properties(self._dataset_requirements)
+        self._stopwatch.start_task(experiment_task_name)
+        self.dataset_name = dataset.dataset_name
+        self.resampling_strategy = dataset.resampling_strategy
+        self._logger = self._get_logger(self.dataset_name)
+        self._disable_file_output = disable_file_output
+        # Save start time to backend
+        self._backend.save_start_time(str(self.seed))
+
+        self._backend.save_datamanager(dataset)
+
+        self._metric = get_metrics(
+            names=[optimize_metric], dataset_properties=dataset_properties)[0]
+
+        self.search_space = get_configuration_space(info=dataset_properties,
+                                                    include=self.include_components,
+                                                    exclude=self.exclude_components)
+        budget_config: Dict[str, Union[float, str]] = {}
+        if budget_type is not None and budget is not None:
+            budget_config['budget_type'] = budget_type
+            budget_config[budget_type] = budget
+        elif budget_type is not None or budget is not None:
+            raise ValueError(
+                "budget was mentioned but not the budget_type to be used with it"
+            )
+
+        if self.task_type is None:
+            raise ValueError("Cannot interpret task type from the dataset")
+
+        self._create_dask_client()
+        proc_ensemble = None
+        if self.ensemble_size <= 0:
```

Comment:
This is  handel a bit differently (see [this](https://github.com/LMZimmer/Auto-PyTorch_refactor/blob/4e2a1b507e930acec4afe03d2f85a39693f4ebea/autoPyTorch/api/base_task.py#L409)). So in this case we don't want to spend precious iterations doing ensemble selection when we already know which is the best model from smac. It is best to lest the above function select the best individual model in case ensemble size is 1 (My argument is also follows from the fact that ensemble selection calculates the score of the predictions which is the most time-consuming operation in the process, something we should consider addressing in the future, because of the many checks done in sklearn).

Reply:
perfect :)